CMakeLists.txt

```diff
@@ -447,7 +470,7 @@ if (NOT NOCRYPTOMINISAT)
     message(STATUS "CryptoMiniSat5 include dirs: ${CRYPTOMINISAT5_INCLUDE_DIRS}")
     add_definitions(-DUSE_CRYPTOMINISAT)
     set(USE_CRYPTOMINISAT ON)
-  else()
+  elseif(NOT SUPRESS_WARNINGS)
```

Comment:
Okay, so we were trying to address *compiler* warnings, not CMake `messages` -- why do you care about turning-off CMake messages?

Reply:
If I add the project via FetchContent, I will see strange warnings from your library. For example "Testing is disabled". But in my project testing is enabled. It just confusing for developers.
I would either modernize the messages (it shouldn't be done like this) or add the option to suppress them.

Reply:
> I would either modernize the messages (it shouldn't be done like this)
>

How are you proposing it should be done?

Reply:
How about [FeatureSummary](https://cmake.org/cmake/help/latest/module/FeatureSummary.html)?

Reply:
If you're happy to do this, it seems like a *great* addition!

For now, can we not do the above (i.e., silencing the CMake warnings) in this PR?

Reply:
> If you're happy to do this, it seems like a great addition!

Yes, I already working on it.

> For now, can we not do the above (i.e., silencing the CMake warnings) in this PR?

Do you want me to send a separate PR for it?

Reply:
> Do you want me to send a separate PR for it?

I would say "yes", because it makes my work easier (it is easier to review and merge smaller, self-contained PRs).

However, if merging this PR is "useless" to you without having `FeatureSummary`, then it likely needs to go in this one.

Or, we finish this one, merge it knowing the job is only half-done, and then merge the `FeatureSummary` one later on. I like this least because there's a chance you never make the second PR, leaving things in an "inconsistent" state.

I should say one thing: I'd rather **not** merge the PR that suppresses the CMake warnings. So either we back-out the change this comment is about, or you need to do the `FeatureSummary` bit.

Reply:
> I would say "yes", because it makes my work easier (it is easier to review and merge smaller, self-contained PRs).

If it easier for you - will do this way.